core/modules/replica/Configuration.cc

```diff
@@ -316,70 +322,114 @@ DatabaseInfo const Configuration::databaseInfo(std::string const& name) const {
-    LOGS(_log, LOG_LVL_DEBUG, context() << "_jobHeartbeatTimeoutSec:             " << _jobHeartbeatTimeoutSec);
-    LOGS(_log, LOG_LVL_DEBUG, context() << "_xrootdAutoNotify:                   " << (_xrootdAutoNotify ? "true" : "false"));
-    LOGS(_log, LOG_LVL_DEBUG, context() << "_xrootdHost:                         " << _xrootdHost);
-    LOGS(_log, LOG_LVL_DEBUG, context() << "_xrootdPort:                         " << _xrootdPort);
-    LOGS(_log, LOG_LVL_DEBUG, context() << "_xrootdTimeoutSec:                   " << _xrootdTimeoutSec);
-    LOGS(_log, LOG_LVL_DEBUG, context() << "_workerTechnology:                   " << _workerTechnology);
-    LOGS(_log, LOG_LVL_DEBUG, context() << "_workerNumProcessingThreads:         " << _workerNumProcessingThreads);
-    LOGS(_log, LOG_LVL_DEBUG, context() << "_fsNumProcessingThreads:             " << _fsNumProcessingThreads);
-    LOGS(_log, LOG_LVL_DEBUG, context() << "_workerFsBufferSizeBytes:            " << _workerFsBufferSizeBytes);
-    LOGS(_log, LOG_LVL_DEBUG, context() << "_databaseTechnology:                 " << _databaseTechnology);
-    LOGS(_log, LOG_LVL_DEBUG, context() << "_databaseHost:                       " << _databaseHost);
-    LOGS(_log, LOG_LVL_DEBUG, context() << "_databasePort:                       " << _databasePort);
-    LOGS(_log, LOG_LVL_DEBUG, context() << "_databaseUser:                       " << _databaseUser);
-    LOGS(_log, LOG_LVL_DEBUG, context() << "_databasePassword:                   " << "*****");
-    LOGS(_log, LOG_LVL_DEBUG, context() << "_databaseName:                       " << _databaseName);
+bool Configuration::setDatabaseAllowReconnect(bool value) {
+    std::swap(value, defaultDatabaseAllowReconnect);
+    return value;
+}
+
+unsigned int Configuration::setDatabaseConnectTimeoutSec(unsigned int value) {
+    if (0 == value) {
+        throw std::invalid_argument(
+                "Configuration::setDatabaseConnectTimeoutSec:  0 is not allowed");
+    }
+    std::swap(value, defaultDatabaseConnectTimeoutSec);
+    return value;
+}
+
+unsigned int Configuration::setDatabaseMaxReconnects(unsigned int value) {
+    if (0 == value) {
+        throw std::invalid_argument(
+                "Configuration::setDatabaseMaxReconnects:  0 is not allowed");
+    }
+    std::swap(value, defaultDatabaseMaxReconnects);
+    return value;
+}
+
+unsigned int Configuration::setDatabaseTransactionTimeoutSec(unsigned int value) {
+    if (0 == value) {
+        throw std::invalid_argument(
+                "Configuration::setDatabaseTransactionTimeoutSec:  0 is not allowed");
+    }
+    std::swap(value, defaultDatabaseTransactionTimeoutSec);
+    return value;
+}
+
+void Configuration::dumpIntoLogger() const {
+    LOGS(_log, LOG_LVL_DEBUG, asString());
+}
+
+std::string Configuration::asString() const {
+
+    std::ostringstream ss;
+    ss << context() << "defaultRequestBufferSizeBytes:        " << defaultRequestBufferSizeBytes << "\n";
```

Comment:
std::endl instead of \n?

Reply:
Probably this doesn't matter here because this is the string-stream class which updates its state each time the streaming operator '\<\<' is used. I can still do what you're asking for.

Reply:
either way, it was just a question.